pareto/strategic_water_management/strategic_produced_water_optimization.py

```diff
@@ -1676,6 +1719,48 @@ def create_model(df_sets, df_parameters, default={}):
             units=model.model_units["pipe_cost_capacity"],
             doc="Pipeline construction/expansion capital cost for selected increment [currency/volume/time]",
         )
+
+    _df_parameters = {}
+    if model.config.hydraulics == Hydraulics.post_process:
+        # determine the max change in elevation from the elevation data
+        max_elevation = max(
+            [abs(val) for val in model.df_parameters["Elevation"].values()]
+        )
+        min_elevation = min(
+            [abs(val) for val in model.df_parameters["Elevation"].values()]
+        )
+        max_elevation_change = max_elevation - min_elevation
+        # Set economic penalties for pipeline operational Cost based on the elevation changes
+        for k1 in model.s_L:
+            for k2 in model.s_L:
+                if (k1, k2) in model.s_LLA:
+                    elevation_delta = value(model.p_Elevation[k1]) - value(
+                        model.p_Elevation[k2]
+                    )
+                    _df_parameters[(k1, k2)] = max(
+                        0,
+                        (
+                            model.df_parameters["PipelineOperationalCost"][(k1, k2)]
+                            - (
+                                0.01
+                                * max(
+                                    [
+                                        abs(val)
+                                        for val in model.df_parameters[
+                                            "PipelineOperationalCost"
+                                        ].values()
+                                    ]
+                                )
+                                * elevation_delta
+                                / max_elevation_change
+                            )
+                        ),
+                    )
+    else:
+        for k1 in model.s_L:
+            for k2 in model.s_L:
+                _df_parameters[(k1, k2)] = 0.0001
```

Comment:
Travis' comment at line 1872 is a good point. It looks like the input data is not being used for False and co optimize config settings. 
_df_parameters (or new variable name) should be set the same way we used to initialize the operating piping cost. 
{
            key: pyunits.convert_value(
                value,
                from_units=model.user_units["currency_volume"],
                to_units=model.model_units["currency_volume"],
            )
            for key, value in model.df_parameters["PipelineOperationalCost"].items()

Reply:
thanks, i agree. i updated it with `_df_parameters = model.df_parameters["PipelineOperationalCost"]`

Reply:
Looks good to me now. Resolving.